--- a/tools/releasetools/sign_target_files_apks.py
+++ b/tools/releasetools/sign_target_files_apks.py
@@ -567,8 +567,9 @@
         print("           : %-*s payload   (%s)" % (
             maxsize, name, payload_key))
 
+        avbtool = misc_info['avb_avbtool'] if 'avb_avbtool' in misc_info else 'avbtool'
         signed_apex = apex_utils.SignApex(
-            misc_info['avb_avbtool'],
+            avbtool,
             data,
             payload_key,
             container_key,
@@ -663,8 +664,9 @@
       # key is specified via --avb_system_other_key.
       signing_key = OPTIONS.avb_keys.get("system_other")
       if signing_key:
+        avbtool = misc_info['avb_avbtool'] if 'avb_avbtool' in misc_info else 'avbtool'
         public_key = common.ExtractAvbPublicKey(
-            misc_info['avb_avbtool'], signing_key)
+            avbtool, signing_key)
         print("    Rewriting AVB public key of system_other in /product")
         common.ZipWrite(output_tf_zip, public_key, filename)
 
